xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiAction.java

```diff
@@ -743,33 +744,16 @@ public void execute(XWikiContext context) throws Exception
     private boolean shouldReturnDocDoesNotExist(XWikiDocument doc, XWikiContext context)
     {
         boolean result = false;
-        if (doc.isNew()) {
-            String action = context.getAction();
-            XWikiRequest request = context.getRequest();
-            if ("view".equals(action)) {
-                result = !"recyclebin".equals(request.get("viewer"))
-                    && !"children".equals(request.get("viewer"))
-                    && !"siblings".equals(request.get("viewer"));
-            } else if ("get".equals(action)) {
-                String sheet = request.get("sheet");
-                if (!StringUtils.isEmpty(sheet)) {
-                    DocumentReference sheetReference = getCurrentMixedDocumentReferenceResolver().resolve(sheet);
-                    try {
-                        XWikiDocument sheetDoc = context.getWiki().getDocument(sheetReference, context);
-                        result = sheetDoc.isNew();
-                    } catch (XWikiException e) {
-                        LOGGER.warn("Error while trying to load sheet [{}] for checking status code "
-                            + "on GET request for [{}]: [{}]", sheetReference, doc.getDocumentReference(),
-                            ExceptionUtils.getRootCauseMessage(e));
-                        // there is an error we consider that the sheet doesn't exist.
-                        result = true;
-                    }
-                } else {
-                    result = !"1".equals(request.get("disableCheckNotExisting"));
-                }
+        String action = context.getAction();
+        if (this.componentManager.hasComponent(DocExistValidator.class, action)) {
+            try {
+                result = this.componentManager.<DocExistValidator>getInstance(DocExistValidator.class, action)
+                    .docExist(doc, context);
+            } catch (ComponentLookupException e) {
+                // Should never happen as we check for the component existence before calling getInstance.
```

Comment:
As I indicated in my previous comment, this would mean that this component could not be initialized, which can definitely happen and which should not be ignored. The minimum is an error log.